### PR TITLE
update bbknn arguments and docstring

### PR DESCRIPTION
**Pull request by ktpolanski**
Updated `sc.external.pp.bbknn()` to match current arguments and docstring.

### CODE REVIEW

**Comment by ktpolanski** [comment]
It seems this is of some relevance to users, as shown by it showing up twice independently over the course of the past week.

For some reason, my tweaks have killed off ReadTheDocs, and I can't check why as upon pressing the "details" button I get the 404 equivalent :) I am now importing `types` so I can correctly define `metric` as also including `types.FunctionType`. This  is probably what is causing whatever hiccup is happening.

@giovp , some assistance with getting the ball rolling on this? The changes are a result of me expanding BBKNN with pynndescent on your recommendation.

**Comment by pinin4fjords** [comment]
@ktpolanski following from https://scanpy.readthedocs.io/en/stable/dev/code.html#code-style and building the docs locally gives me (among some errors that were not relevant):

```
...
/path/to/scanpy/scanpy/external/pp/_bbknn.py:docstring of scanpy.external.pp._bbknn.bbknn:28: WARNING: py:class reference target not found: function
/path/to/scanpy/scanpy/external/pp/_bbknn.py:docstring of scanpy.external.pp._bbknn.bbknn:28: WARNING: py:class reference target not found: function
...
```

Maybe that gives you some pointers?

**Comment by ktpolanski** [comment]
It does not, as an analogous docstring built just fine over on the [BBKNN page](https://bbknn.readthedocs.io/en/latest/).

**Comment by ktpolanski** [comment]
I ran `pip3 install scanpy[doc]`, as instructed by that page, and the doc seems to have built fine locally.

image

The only fishy thing is the `function` is not clickable, while the other two are, but it built.

Also, line 28 is `trim: Optional[int] = None,`, which was there previously already. It's the metric change that's causing this - your argument renaming PR did not hiccup.

**Comment by pinin4fjords** [comment]
Hmm, I'm stumped. @ivirshup ?

File: scanpy/external/pp/_bbknn.py [hunk 1]
```diff
@@ -14,13 +15,20 @@
 def bbknn(
     adata: AnnData,
     batch_key: str = 'batch',
+    use_rep: str = 'X_pca',
```
Comment (ivirshup) [inline scanpy/external/pp/_bbknn.py:17]: Could the representation be chosen with keywords like `obsm`, `layers`? You can use the helper functions `sc.get._get_obs_rep` and `sc .get._set_obs_rep`.

**Review by ivirshup — COMMENTED** [review]
> The only fishy thing is the function is not clickable, while the other two are, but it built

I think this is it. The error I see is:

```
sphinx.errors.SphinxWarning: /home/docs/checkouts/readthedocs.org/user_builds/icb-scanpy/checkouts/1868/scanpy/external/pp/_bbknn.py:docstring of scanpy.external.pp._bbknn.bbknn:28:py:class reference target not found: function

Warning, treated as error:
/home/docs/checkouts/readthedocs.org/user_builds/icb-scanpy/checkouts/1868/scanpy/external/pp/_bbknn.py:docstring of scanpy.external.pp._bbknn.bbknn:28:py:class reference target not found: function
```

I don't think the `'types.FunctionType'` needs to be quoted, but we've used `collections.abc.Callable` in other places without issue.

**Comment by ktpolanski** [comment]
I can't get it to work. I have unquoted `types.FunctionType`, and `function` remains unclickable. I have tried to switch to an `collections.abc.Callable`, and instead of anything helping the entire argument support vanishes:

image

Please help?

**Comment by ivirshup** [comment]
I think that should do it. I believe the `types` module isn't actually meant to be used for type annotations, that's what the `typing` module is for (which is always confusing to me).

> and instead of anything helping the entire argument support vanishes:

I got this behaviour too, but running `make clean` and building from scratch got things working again.  ¯\_(ツ)_/¯

**Comment by ktpolanski** [comment]
Thanks a lot for the help, assuming things are in order then that's it for me.

**Comment by ivirshup** [comment]
Thanks for the update!

**Pull request merged**